typescript/audio-summarizer/README.md

```diff
@@ -0,0 +1,188 @@
+- Copy my prompt from the code to the editor
+- Adjust models, temperatures and other settings as needed
+- Save the prompt with the name "Emojifier Prompt" (be sure to copy this exact name as it will be used to retrieve the prompt through the API)
+
+Now I will edit the `src/app/api/emojify/route.ts` file and add the following :
+
+```ts
+import { LiteralClient } from "@literalai/client";
+
+const openai = new OpenAI();
+// Init the Literal client
+const literalClient = new LiteralClient();
+
+export async function POST(req: NextRequest) {
+  // ...
+
+  // Get the prompt from the Literal API
+  const promptName = "Emojifier Prompt";
+  const prompt = await literalClient.api.getPrompt(promptName);
+  if (!prompt) throw new Error("Prompt not found");
+  const promptMessages = prompt.formatMessages();
+
+  // Call the LLM API
+  const completion = await openai.chat.completions.create({
+    ...prompt.settings,
+    messages: [...promptMessages, { role: "user", content: text }],
+  });
+
+  // ...
+}
+```
+
+## Logging the requests
+
+I want to log each request as a [Run](https://docs.getliteral.ai/concepts/observability/run) which will contain two [steps](https://docs.getliteral.ai/concepts/observability/step) :
+
+- One step for the audio transcription
+- One step for the summarization
+
+Here is what it will look like on Literal AI :
+
+[![Literal AI Run]](01-requests-logging.png)
+
+### Logging the run
+
+To facilitate its use, i will generate the Run ID from the frontend using `crypto.randomUUID()` and pass it to the backend. This ensures that my run IDs are unique and fully compatible with Literal AI. I then simply add `runId` to the payload of the API requests.
+
+In the `src/app/api/transcribe/route.ts` I will then create a [thread](https://docs.getliteral.ai/concepts/observability/thread) for each run. This is a bit of a hack as the interaction is not really a threaded conversation, however it is necessary so that I can upload Audio files.
+
+```ts
+// We receive the file from the frontend as a Blob, however Literal expects a ReadableStream so we have to convert it
+const nodeStream = Readable.fromWeb(formAudio.stream() as ReadableStream<any>);
+
+// Upload the file to Literal and add it as an attachment
+const { objectKey } = await literalClient.api.uploadFile({
+  content: nodeStream,
+  threadId: thread.id,
+  mime: "audio/webm",
+});
+const attachment = new Attachment({
+  name: "Audio file",
+  objectKey,
+  mime: "audio/webm",
+});
+
+// Create the run with the attached audio file
+const run = await thread
+  .step({
+    id: runId,
+    type: "run",
+    name: "Emojifier 🤖🎨",
+    input: {
+      input: { content: "Audio file" },
+      attachments: [attachment],
+    },
+  })
+  .send();
+```
+
+### Logging the steps
+
+Still in `src/app/api/transcribe/route.ts`, I then add the first step for the audio transcription. Please note that I am measuring start and end time, which will allow me to monitor latency from Literal AI.
+
```

Comment:
Doesn't Literal AI measure the latency itself?

Reply:
On regular ChatGPT calls we have the openAI instrumentation which handles the latency, token counts etc... However this does not apply to the calls to the Whisper speech-to-text API so this has to be done manually. Do you think i should explicit that ?